lib/core/src/data/weakset.rs

```diff
@@ -0,0 +1,113 @@
+use std::collections::HashSet;
+use std::rc::{Rc, Weak};
+use std::hash::{Hash, Hasher};
+pub trait Hashable = Hash + PartialEq;
+
+pub struct WeakElem<T : Hashable> {
+    elem : Weak<T>
+}
+
+pub struct WeakRef<T : Hashable> {
+    elem : Rc<T>,
+    owner : Rc<WeakSet<T>>
+}
+
+impl<T : Hashable> Drop for WeakRef<T> {
+    fn drop(&mut self) {
+        unsafe {
+            // no worries because it will happen synchronously
```

Comment:
Please use style guide regarding comments in code. Plus, this comment is not specific enough - comments regarding `unsafe` blocks should be really detailed and explain why we need such blocks.

Reply:
weakset.rs was removed as suggested